src/main/java/io/spine/helloworld/client/Client.java

```diff
@@ -88,6 +89,24 @@ private void printEvent(EventMessage e) {
         System.out.println(out);
     }
 
+    /**
+     * Cancels all current subscriptions, if any.
+     */
+    private void cancelSubscriptions() {
+        if (subscriptions != null) {
+            subscriptions.forEach(s -> client.subscriptions().cancel(s));
```

Comment:
Can we extract the `client.subscriptions()` call result into a variable and then use a method reference here?

```
Subscriptions subscriptions = client.subscriptions();
this.subscriptions.forEach(subscriptions::cancel);
```

Reply:
I've tried but it looks a bit like a programming puzzle: Which subscriptions cancel which?

```java
    private void cancelSubscriptions() {
        if (subscriptions != null) {
            Subscriptions subscriptions = client.subscriptions();
            this.subscriptions.forEach(subscriptions::cancel);
            this.subscriptions = null;
        }
    }
```